src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,261 @@
+"""
+Extraction algorithms to compute the statistics from a chunk of images
+"""
+
+__all__ = [
+    "StatisticsExtractor",
+    "PlainExtractor",
+    "SigmaClippingExtractor",
+]
+
+from abc import abstractmethod
+
+import numpy as np
+from astropy.stats import sigma_clipped_stats
+
+from ctapipe.containers import StatisticsContainer
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import (
+    Int,
+    List,
+)
+
+
+class StatisticsExtractor(TelescopeComponent):
+    """
+    Base component to handle the extraction of the statistics from a dl1 table
+    containing charges, peak times and/or charge variances (images).
+    """
+
+    chunk_size = Int(
+        2500,
+        help="Size of the chunk used for the calculation of the statistical values",
+    ).tag(config=True)
+
+    def __call__(
```

Comment:
I think for the base class, `__call__` should be abstract and the arguments of `__call__` need to be the same for all subclasses to be exchangeble via the configuration system without adapting the call site.

Reply:
Maybe I'm missing something, but the `__call__` is designed to be identical for all the subclasses. Simply the function `extract()` is implemented in each subclass.